src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -149,6 +201,8 @@ public void removeTaskId(String taskId) {
         if (entry.getValue() != null && entry.getValue().equals(taskId)) {
           journalNodes.put(entry.getKey(), null);
           setJournalNodes(journalNodes);
+          Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+          deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
That is true. However, this adds complexity to the code and this is a rare case of a DN failing and restarting on the same host over and over again. Perhaps in this case, we don't want to start new datanodes and want to take down that host instead in the case of multiple failures.